scalr/resource_scalr_policy_group.go

```diff
@@ -108,15 +108,38 @@ func resourceScalrPolicyGroup() *schema.Resource {
 				},
 			},
 			"environments": {
-				Description: "A list of the environments the policy group is linked to.",
+				Description: "A list of the environments the policy group is linked to. Use `[\"*\"]` to enforce in all environments.",
 				Type:        schema.TypeList,
-				Computed:    true,
+				Optional:    true,
 				Elem:        &schema.Schema{Type: schema.TypeString},
 			},
 		},
 	}
 }
 
+func parseEnvironmentDefinitions(d *schema.ResourceData) ([]*scalr.Environment, error) {
+	environmentIDs := d.Get("environments").([]interface{})
+	err := ValidateIDsDefinitions(environmentIDs)
+	if err != nil {
+		return nil, fmt.Errorf("got error during parsing environments: %s", err.Error())
+	}
+
+	environments := make([]*scalr.Environment, 0)
+	var hasWildcard bool
+	for i, environmentID := range environmentIDs {
+		environments = append(environments, &scalr.Environment{ID: environmentID.(string)})
+
+		if environmentID.(string) == "*" {
+			hasWildcard = true
+		}
+		if i != 0 && hasWildcard {
```

Comment:
So this allows the list to be `["*", "env-xxx", "env-yyy"]`?
I find the logic in [this example](https://github.com/Scalr/terraform-provider-scalr/blob/880edf59a8c17a39c0f6e0e8128c02703cf8f9b3/scalr/resource_scalr_vcs_provider.go#L233) more clear and correct.

Reply:
logic is correct, but changed it to be consistent with existing ones